src/crt/sllshl.src

```diff
@@ -0,0 +1,125 @@
+; Performs 48 bit ("short long long") bit shift left
+; 
+; Arguments:
+; ude = Most significant bytes of arg 1
+; uhl = Least significant bytes of arg 1
+; ubc = Amount to shift by (effectively just shift by c)
+
+
+; Returns:
+; ude:uhl = ude:uhl<<c
+
+assume	adl=1
+
+	section	.text
+	public	__sllshl
+__sllshl:
+	push iy
+	push de
+	push hl
+	pop iy ; iy will be used for least significant, hl for most going forward
+	pop hl
+
+	bit	0, c
+	jr	z, .bit0
+	add	iy, iy
+	adc hl, hl
+.bit0:
+	bit	1, c
+	jr	z, .bit1
+	add	iy, iy
+	adc hl, hl
+	add	iy, iy
+	adc hl, hl
+.bit1:
+	bit	2, c
+	jr	z, .bit2
+	add	iy, iy
+	adc hl, hl
+	add	iy, iy
+	adc hl, hl
+	add	iy, iy
+	adc hl, hl
+	add	iy, iy
+	adc hl, hl
+.bit2:
+	bit	3, c
+	jr	z, .bit3
+	add	iy, iy
```

Comment:
Unrolling a left shift by 8 like this may be fast if it's just HL, but across two registers like this, it's slower (and far more bloated) than the stack-based equivalent. The same goes for the other byte-aligned shifts below as well.